runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/AbstractDokkaParentTask.kt

```diff
@@ -0,0 +1,86 @@
+package org.jetbrains.dokka.gradle
+
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Nested
+import org.jetbrains.dokka.DokkaBootstrap
+import org.jetbrains.dokka.DokkaBootstrapImpl
+import kotlin.reflect.KClass
+
+abstract class AbstractDokkaParentTask(
+    bootstrapClass: KClass<out DokkaBootstrap> = DokkaBootstrapImpl::class
+) : AbstractDokkaTask(bootstrapClass) {
+
+    @get:Internal
+    internal var childDokkaTaskPaths: Set<String> = emptySet()
+        private set
+
+    @get:Nested
+    internal val childDokkaTasks: Set<AbstractDokkaTask>
```

Comment:
A task inside a task does not seem right. Why is this needed?

Reply:
This kind of "parent" tasks are managing multiple "child" tasks. Accessing them like this is not just there for convenience. While not all parent tasks do have task dependencies on its children, they should still re-run if any of the child's configuration changes. This happens to work pretty nicely with this `@Nested` pointing to the actual tasks ☺️